pkg/transfer/local/pull_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package local
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/pkg/transfer"
+	"github.com/containerd/containerd/pkg/unpack"
+	"github.com/containerd/containerd/platforms"
+)
+
+func TestGetSupportedPlatform(t *testing.T) {
+	supportedPlatforms := []unpack.Platform{
+		{
+			Platform:       platforms.OnlyStrict(platforms.MustParse("linux/amd64")),
+			SnapshotterKey: "native",
+		},
+		{
+			Platform:       platforms.OnlyStrict(platforms.MustParse("linux/amd64")),
+			SnapshotterKey: containerd.DefaultSnapshotter,
```

Comment:
suggest to use `overlayfs` to maintain easier here because the `DefaultSnapshotter` in windows can be `windows` which is kind of confuse.

Reply:
I didn't think about Windows earlier. I agree if the test runs on Windows, it doesn't make sense for a combination of Linux platform with `windows` snapshotter. 
I created a new supportPlatform [entry](https://github.com/containerd/containerd/pull/8165/files#diff-3e88bd9486ed6f469c1a97bd4ab513533167b86bcceceda7c6496c624fe88ac0R47) which is the combination of `defaultPlatform` and `defaultSnapshotter` to make it platform independent. It makes test case [Platform input only match defaultSnapshotter](https://github.com/containerd/containerd/pull/8165/files#diff-3e88bd9486ed6f469c1a97bd4ab513533167b86bcceceda7c6496c624fe88ac0R111) easier. 
Let me know whether this makes sense to you.